src/Bicep.Core/TypeSystem/Providers/ResourceTypeProviderFactory.cs

```diff
@@ -25,7 +28,7 @@ public class ResourceTypeProviderFactory : IResourceTypeProviderFactory
             = new(() => new AzResourceTypeProvider(new AzResourceTypeLoader(new AzTypeLoader()), AzNamespaceType.Settings.ArmTemplateProviderVersion));
 
         private record ResourceTypeLoaderKey(string Name, string Version);
-        private readonly Dictionary<ResourceTypeLoaderKey, IResourceTypeProvider> cachedResourceTypeLoaders = new();
+        private readonly ConcurrentDictionary<ResourceTypeLoaderKey, IResourceTypeProvider> cachedResourceTypeLoaders = new();
```

Comment:
Are you expecting multiple threads at the same time?

Reply:
It's definitely a possibility that there would be multiple threads needing access to this data structure, Anthony recommended in a previous comment to use a thread safe one